### ISSUE TITLE
stale element reference from site_prism 2.15.1

### ISSUE THREAD

**Issue by mathieujobin** [site-prism/site_prism#53]
# stale element reference from site_prism 2.15.1

problem upgrading site_prism from 2.14 to 2.15

I don't have this problem with 2.14.
bumping this gem is the only changes I have

## Environment
- Operating System Ubuntu 18.04 + 20.04
- site_prism 2.15.1 and 2.16 behave the same, 2.14 is working and `capybara` 2.18 / selenium-webdriver (3.142.7)
- How you're interacting with the Framework:  *RSpec Units*
- Browser/s Driver/s being used - *Chrome 81*

## Expected Behavior
elements should be found, test should pass.

## Actual Behavior

when I access the `rules` section, trying to simply `expect(dialog.rules.length).to eq(2)`

I get this error.
```
*** Selenium::WebDriver::Error::StaleElementReferenceError Exception: stale element reference: element is not attached to the page document
  (Session info: chrome=81.0.4044.138)
```

## Steps to reproduce

```
  class EditThresholdRule < SitePrism::Section
    element :condition_type, "select[name='condition_type']"
    element :action_type, "select[name='action_type']"
    element :action_options_rounding, "select[name='action_options_rounding']"
    element :action_options_minutes, "select[name='action_options_minutes']"
    element :destroy, "a.destroy_link"
  end
  class EditThresholdDialog < UiDialog
    element :save_and_apply, "button.primary-button"
    element :select_allow, "select[name='allow']"
    sections :rules, EditThresholdRule, "li.resolution-rules li.rule"
    element :add_rule, "li.resolution-rules li.actions button"
  end
```

probably important to add, these elements are hidden/shown via angular, but I see them in the browser when I stop the debugger

thanks

**Comment by luke-hill** [comment]
There isn't enough information here to reproduce. Ideally follow the guide at SSCCE.org

**Comment by mathieujobin** [comment]
thank you, I marked two of my tests as pending so I am able to move forward, and 1 of them fixed itself when I upgraded to 3.0
